server/src/test/java/org/spine3/server/storage/ProjectionStorageShould.java

```diff
@@ -69,6 +85,208 @@ public void return_null_if_no_event_time_in_storage() {
+
+        final FieldMask nonEmptyMask = FieldMask.newBuilder()
+                                                .addPaths("invalid_path")
+                                                .build();
+        final Map<I, EntityStorageRecord> masked = storage.readAll(nonEmptyMask);
+
+        assertEmpty(noMask);
+        assertEmpty(masked);
+
+        // Same type
+        assertEquals(noMask, masked);
+    }
+
+    @SuppressWarnings({"MethodWithMultipleLoops", "BreakStatement"})
+    @Test
+    public void perform_read_bulk_operations() {
+        final int count = 10;
+        final List<I> ids = new LinkedList<>();
+
+        for (int i = 0; i < count; i++) {
+            final I id = newId();
+            final ProjectId projectId = ProjectId.newBuilder()
+                                                 .setId(id.toString())
+                                                 .build();
+            final Project state = Project.newBuilder()
+                                         .setId(projectId)
+                                         .build();
+
+            final Any packedState = AnyPacker.pack(state);
+
+            final EntityStorageRecord record = EntityStorageRecord.newBuilder()
+                                                                  .setState(packedState)
+                                                                  .setWhenModified(Timestamps.getCurrentTime())
+                                                                  .setVersion(1)
+                                                                  .build();
+            storage.write(id, record);
+
+            // Request a subset of records
+            if (i % 2 == 0) {
+                ids.add(id);
+            }
+        }
+
+        final Iterable<EntityStorageRecord> read = storage.readBulk(ids);
+        assertSize(ids.size(), read);
+
+        // Check data consistency
+        for (EntityStorageRecord record : read) {
+            final Any packedState = record.getState();
+            final Project state = AnyPacker.unpack(packedState);
+            final ProjectId id = state.getId();
+            final String stringIdRepr = id.getId();
+
+            boolean isIdPresent = false;
+            for (I genericId : ids) {
+                isIdPresent = genericId.toString()
+                                       .equals(stringIdRepr);
+                if (isIdPresent) {
+                    break;
+                }
+            }
+            assertTrue(isIdPresent);
+        }
+    }
+
+    @SuppressWarnings({"MethodWithMultipleLoops", "BreakStatement"})
+    @Test
+    public void perform_bulk_read_with_field_mask_operation() {
+        final int count = 10;
+        final List<I> ids = new LinkedList<>();
+
+        final String projectDescriptor = Project.getDescriptor()
+                                                .getFullName();
+        final FieldMask fieldMask = FieldMask.newBuilder()
```

Comment:
Consider extracting some bits of this method:

* they look like reusable in other test methods;
* you'll make the code of this method more readable.